lib/index.js

```diff
@@ -11,22 +11,31 @@ module.exports = (server, options, next) => {
 
     server.decorate('reply', 'push', function (response, path, headers) {
 
-        if (typeof response === 'string') {
+        if (typeof response === 'string' || Array.isArray(response)) {
             headers = path;
             path = response;
             response = this.request.response;
         }
 
         Hoek.assert(response && !response.isBoom, 'Must server-push with a non-error response.');
+        Hoek.assert(!!path, 'Must provide a path to the resource to be pushed.');
+
+        // now we force path to be an Array (makes life easier later)
+        if (!Array.isArray(path)) {
+            path = [path];
+        }
 
         headers = headers || {};
 
-        // Adjust for relative path
-        path = (path[0] === '/') ? path : `${this.realm.modifiers.route.prefix || ''}/${path}`;
+        const pushes = path.map((path) => {
```

Comment:
Mind using a for loop here?  I realize it's sort of a micro-optimization, but I try to keep close to hapi's best practices, which suggest using for loops in hot code paths like route handlers.  Rest assured I use `map()` elsewhere liberally!